Source/JavaScriptCore/dfg/DFGBackwardsPropagationPhase.cpp

```diff
@@ -547,20 +558,20 @@ class BackwardsPropagationPhase {
                 // then -0 and 0 are treated the same.  We don't need NodeBytecodeUsesAsOther
                 // because if all of the cases are integers then NaN and undefined are
                 // treated the same (i.e. they will take default).
-                node->child1()->mergeFlags(NodeBytecodeUsesAsNumber | NodeBytecodeUsesAsInt);
+                node->child1()->mergeFlags(NodeBytecodeUsesAsNumber | NodeBytecodeUsesAsInt | NodeBytecodeNeedsNaNOrInfinity);
```

Comment:
Is this needed?

Reply:
Yes, NaN and 0 is different in `switch(imm)` context.